packages/sdk/src/evm/functions/contract/index.ts

```diff
@@ -0,0 +1,102 @@
+import { getCompositePluginABI } from "../../common/plugin";
+import { PREBUILT_CONTRACTS_MAP } from "../../contracts";
+import { SmartContract } from "../../contracts/smart-contract";
+import { ContractPublisher } from "../../core/classes/contract-publisher";
+import { getSignerAndProvider } from "../../core/classes/rpc-connection-handler";
+import { ValidContractInstance } from "../../core/types";
+import { AbiSchema } from "../../schema/contracts/custom";
+import {
+  cacheContract,
+  getCachedContract,
+  getCachedStorage,
+  inContractCache,
+} from "../core/cache";
+import { GetContractFromAbiParams, GetContractParams } from "./types";
+import { resolveContractType } from "./utils";
+
+export async function getContractFromAbi(
+  params: GetContractFromAbiParams,
+): Promise<ValidContractInstance> {
+  if (inContractCache(params.address)) {
```

Comment:
I believe this needs to include more than just the address, otherwise you may get the same contract instance back for contract `0x123...` and `0x123...` if they're on different chains, so it needs to include at least chainId + address

Reply:
Indeed